ui/src/components/admin/ModelsManager/ModelManagerController.js

```diff
@@ -115,350 +120,356 @@ Model table state transitions for each action:
+                            case VARIANT_OVERWRITE_OPTIONS.none:
+                            default:
+                              break;
+                          }
+                        }),
+                      )
+                      .catch(async err => {
+                        const errorText = extractErrorText(err);
 
-                    await setState({
-                      isLoading: false,
-                    });
+                        await setState({
+                          isLoading: false,
+                        });
 
-                    await appendNotification({
-                      type: NOTIFICATION_TYPES.ERROR,
-                      message: 'Model Upload Error.',
-                      details: errorText.length > 0 ? errorText : 'Unknown error has occurred.',
-                    });
-                  });
-              },
-              bulkImportVariants: async modelNames => {
-                return importBulkGenomicVariants(modelNames)
-                  .then(response => {
-                    if (response.data.success) {
-                      setImportProgress({
-                        ...importProgress,
-                        running: true,
+                        await appendNotification({
+                          type: NOTIFICATION_TYPES.ERROR,
+                          message: 'Model Upload Error.',
+                          details: errorText.length > 0 ? errorText : 'Unknown error has occurred.',
+                        });
+                      });
+                  },
+                  bulkImportVariants: async modelNames => {
+                    return importBulkGenomicVariants(modelNames)
+                      .then(response => {
+                        if (response.data.success) {
+                          setImportProgress({
+                            ...importProgress,
+                            running: true,
+                          });
+                        }
+                      })
+                      .catch(async error => {
+                        await appendNotification({
+                          type: NOTIFICATION_TYPES.ERROR,
+                          message: 'Bulk Import of Research Somatic Variants Failed.',
+                          details: error.response
+                            ? error.response.data.error.message
+                            : error.message,
+                          timeout: false,
+                        });
                       });
-                    }
-                  })
-                  .catch(async error => {
-                    await appendNotification({
-                      type: NOTIFICATION_TYPES.ERROR,
-                      message: 'Bulk Import of Research Somatic Variants Failed.',
-                      details: error.response ? error.response.data.error.message : error.message,
+                  },
+                  bulkPublish: async () => {
```

Comment:
UI bulk publish function changes begin here (line 324) and go to line 354.

Removed locking the UI for synchronous publishing, replaced with call to async publisher.